src/dotnet/Chat.Service/ChatsMigrationBackend.cs

```diff
@@ -0,0 +1,449 @@
+        ChatId newChatId,
+        IReadOnlyCollection<MigratedAuthor> migratedAuthors,
+        CancellationToken cancellationToken)
+    {
+        int updateCount;
+        var chatEntryAuthorSids = await dbContext.ChatEntries
+            .Where(c => c.ChatId == chatSid)
+            .Select(c => c.AuthorId)
+            .Distinct()
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false);
+
+        foreach (var authorSid in chatEntryAuthorSids) {
+            var authorId = new AuthorId(authorSid);
+            AuthorId newAuthorId;
+            if (authorId.LocalId > 0)
+                newAuthorId = migratedAuthors.First(c => OrdinalEquals(c.OriginalAuthor.Id.Value, authorSid)).NewId;
+            else if (authorId.LocalId == Constants.User.Walle.AuthorLocalId)
+                newAuthorId = new AuthorId(newChatId, authorId.LocalId, AssumeValid.Option);
+            else
+                throw StandardError.Constraint($"Unexpected author local id. Local id is '{authorId.LocalId}'.");
+
+            var newAuthorSid = newAuthorId.Value;
+            updateCount = await dbContext.ChatEntries
+                .Where(c => c.ChatId == chatSid)
+                .Where(c => c.AuthorId == authorId)
+                .ExecuteUpdateAsync(s => s.SetProperty(c => c.AuthorId, newAuthorSid), cancellationToken)
+                .ConfigureAwait(false);
+            AssertAtLeastOneUpdated(updateCount);
+
+            Log.LogInformation("Updated AuthorId for {Count} chat entry records with author id '{AuthorId}'." +
+                " New author id is '{NewAuthorId}'", updateCount, authorSid, newAuthorSid);
+        }
+
+        for (int i = 0; i < 2; i++) {
+            var entryKind = (ChatEntryKind)i;
+            var prefix = newChatId + ":" + i + ":";
+            updateCount = await dbContext.ChatEntries
+                .Where(c => c.ChatId == chatSid)
+                .Where(c => c.Kind == entryKind)
+                .ExecuteUpdateAsync(setters => setters
+                        .SetProperty(c => c.Id, c => prefix + c.LocalId)
+                        .SetProperty(c => c.ChatId, newChatId),
+                    cancellationToken)
+                .ConfigureAwait(false);
+
+            Log.LogInformation("Updated Id and ChatId for {Count} '{Kind}' chat entry records", updateCount, entryKind.ToString());
+        }
+
+        // NOTE: I expect that we have not many entries with forward fields filled in so far
+        // hence it's ok to fetch them all at once.
+        var chatEntriesToUpdateForwardFields = await dbContext.ChatEntries
+            .Where(c => c.ForwardedAuthorId != null && c.ForwardedAuthorId.StartsWith(chatSid))
+            .ToListAsync(cancellationToken)
+            .ConfigureAwait(false);
+
+        updateCount = 0;
+        foreach (var dbChatEntry in chatEntriesToUpdateForwardFields) {
+            var forwardedAuthorId = dbChatEntry.ForwardedAuthorId.RequireNonEmpty("ForwardedAuthorId");
```

Comment:
"ForwardedAuthorId" -> nameof(dbChatEntry.ForwardedAuthorId)

Reply:
And the same further.

Reply:
fixed